android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -765,11 +804,21 @@ public void receiveCommand(WebView root, int commandId, @Nullable ReadableArray
   }
 
   @Override
-  public void onDropViewInstance(WebView webView) {
-    super.onDropViewInstance(webView);
-    ((ThemedReactContext) webView.getContext()).removeLifecycleEventListener((RNCWebView) webView);
-    ((RNCWebView) webView).cleanupCallbacksAndDestroy();
-    mWebChromeClient = null;
+  public void onDropViewInstance(RNCWebView wrapper) {
+    super.onDropViewInstance(wrapper);
+
+    // View has already been reattached, do nothing
+    if (wrapper.getWebView() == null) {
+      return;
+    }
+
+    if (wrapper.getWebView().webViewKey == null && !wrapper.getWebView().keepWebViewInstanceAfterUnmount) {
```

Comment:
Should that be an `||` instead of an `&&`? I think both the webview key should be non-null, and `keepWebViewInstanceAfterUnmount` should be true for this to reuse the internal WebView. If either one of those is false, I think we can clean up the internal WebView without removing the internal WebView from its parent

Reply:
Hmm yeah, I think this line is wrong. The way I thought about in my head was this:

1) webViewKey indicates that this webView can be re-used if exists
2) keepWebViewInstanceAfterUnmount indicates that this webview should stay in memory until manually released

Although, I think after reading your comment, my understanding might be a bit incorrect and it seems like the presence of either value indicates that this webView should remain in memory, is that the case (I'll double check the iOS code in a bit)

Reply:
> and it seems like the presence of either value indicates that this webView should remain in memory

I vaguely recall the iOS logic being that we needed both values to be true to keep the webview in memory, but feel free to correct me if I'm wrong after you check the iOS code

Reply:
> iOS logic being that we needed both values to be true to keep the webview in memory

Yup this is right. Do you like having `keepWebViewInstanceAfterUnmount` to be more explicit or was there another reason behind having that field, even though `webViewKey != nil` was enough.

Reply:
If I'm remembering correctly, I think I added both `webViewKey` and `keepWebViewInstanceAfterUnmount` to make it more explicit. It might've been more future-proof / flexible, too, since it'd allow us to use the `webViewKey` for other cases, too, (though I haven't actually thought of any other uses cases for the `webViewKey` yet)

Reply:
I've conformed to iOS for now, when we have other possible use-cases we can discuss at that point I guess